ludwig/train.py

```diff
@@ -396,6 +400,68 @@ def full_train(
     )
 
 
+def kfold_cross_validate(
+                model_definition,
+                model_definition_file=None,
+                data_csv=None,
+                data_train_csv=None,
+                data_validation_csv=None,
+                output_directory='results',
+                k_fold=None,
```

Comment:
I would make this parameter mandatory

Reply:
Couple of questions:

1) I just want to confirm that when you write, "this parameter" you mean the "k_fold" parameter.  Is this correct?

2) Re: the function signature.  Couple of major issues called out by deepsource code quality check identified unused parameters `data_csv` and 'data_validation_csv`.  To resolve those issues, I removed them from the function definition.  Right now I'm assuming that it does not make sense to duplicate all the parameters from `full_train()` into `kfold_cross_validate()`

Reply:
This is the new def function:
```
def kfold_cross_validate(
                k_fold,
                model_definition_file=None,
                data_train_csv=None,
                output_directory='results',
                **kwargs
):
```

A follow-up question, should `model_definition_file` and `data_train_csv` be mandatory as well?

Reply:
In the current API you have model_definition and model_definition_file both None and then there's some logic to chack tha at least one of them is not none: https://github.com/uber/ludwig/blob/master/ludwig/api.py#L141-L156
You could do the same.

Regarding data_train_csv, I would actually call it data_csv. The rationale being that that data is going to be split in folds and used for both training and test, so I'd rather be more generic.